### PR TITLE
Feature/start and end game

### LINKED ISSUE #31: GameController — startGame() & endGame()
Description: Translates the "Start A Game" and "End Game" user stories into code.

- startGame(): prompt for num players, use DeckFactory.buildDeck(), create Player objects, deal hands, initialize GameState
- endGame(): set GameStatus to ENDED, display winner via IGameDisplay, handle reset or close
- Write integration tests covering the Start A Game postconditions (each player has 8 cards, deck has n-1 exploding kittens, discard pile is empty)
- If finished early: write integration tests for the full start-to-first-turn flow

### PR DESCRIPTION
**Pull request by carolineguerraa**
Address #31 
Summary
 - Created docs/bva/ui/GameController.md with boundary value analysis for startGame() and endGame()                                                                               
  - Implemented startGame() in GameController: validates player count (2–5) with re-prompt on invalid input, creates Player objects, initializes GameState                         
  - Implemented endGame() in GameController: sets game status to ENDED, displays winner via IGameDisplay, and delegates restart/close to input.promptRestart()                     
  - Added promptRestart() to IPlayerInput and stubbed it in GameView                                                                                                               
  - Added GameControllerTest with 7 tests covering both methods, written TDD-style (minimum passing code added per test)                                                           
                                                                                                                                                                                   
  Test plan                                                                                                                                                                      
                                                                                                                                                                                   
  - startGame_ValidMinPlayers_InitializesWithoutError — no error shown for numPlayers = 2                                                                                          
  - startGame_ValidMaxPlayers_InitializesWithoutError — no error shown for numPlayers = 5
  - startGame_InvalidNumPlayersBelowMin_ShowsErrorAndRepromptsNumPlayers — re-prompts on numPlayers = 1                                                                            
  - startGame_InvalidNumPlayersAboveMax_ShowsErrorAndRepromptsNumPlayers — re-prompts on numPlayers = 6                                                                            
  - endGame_OneActivePlayer_DisplaysSurvivor — showWinner called after endGame()                                                                                                   
  - endGame_PromptRestartFalse_DoesNotCallStartGame — startGame() not re-entered when restart = false                                                                              
  - endGame_PromptRestartTrue_CallsStartGame — startGame() re-entered when restart = true

### CODE REVIEW

File: src/main/java/ui/GameController.java [hunk 1]
```diff
@@ -1,24 +1,50 @@
...
+	}
+
+	public void startGame() {
+		int numPlayers = input.promptNumPlayers();
+		while (numPlayers < MIN_PLAYERS || numPlayers > MAX_PLAYERS) {
+			display.showMessage("Please enter a number of players between 2 and 5.");
+			numPlayers = input.promptNumPlayers();
+		}
+		List<Player> players = buildPlayers(numPlayers);
+		this.gameState = new GameState(players, new Deck(new ArrayList<>()));
```
Comment (merc-cyy) [inline src/main/java/ui/GameController.java]: use DeckFactory.buildDeck() to build the deck

**Review by merc-cyy — COMMENTED** [review]
added comments

File: src/main/java/ui/GameController.java (hunk 1 above)
Comment (carolineguerraa) [reply src/main/java/ui/GameController.java]: Thanks, fixed!

**Review by merc-cyy — APPROVED** [review]
